db/FetchArchive.py

```diff
@@ -0,0 +1,69 @@
+import requests
+from bs4 import BeautifulSoup as soup
+import json
+
+
+BASE_URL = 'https://archive.org'
+
+
+def archiveIndexGenerator():
+    '''
+    Fetches the index from Archive.org.
+    Contains identifiers used to construct links to episode pages.
+    '''
+    i, n = 0, 0  # Store page_num, segment_count
+    cursor = None  # Token for archive.org to get next page of results.
+    url = BASE_URL + '/services/search/v1/scrape'
+    while i == 0 or cursor is not None:
+        print(f'Fetching segments on page {i}, found {n} segments already')
+        # TODO: Update payload['q'] to fetch other shows.
+        payload = {
+            'q': 'collection:(TV-BLOOMBERG)',
+            'count': 10000,  # 10,000 is the max
+            'fields': 'date,forumSubject,title,identifier',
+            'sorts': 'date',
+            'cursor': cursor,
+        }
+        res = requests.get(url, payload)
+        assert(res.status_code == 200)
+        data = json.loads(res.text)
+        for item in data['items']:
+            yield item
+        cursor = data.get('cursor')
+        i += 1
+        n += len(data['items'])
+    print(f'Found {n} segments')
+
+
+def getEpisode(identifier):
+    '''
+    Makes web requests to archive.org. Scrapes episode data.
+    Returns a dictionary like {metadata: {}, snippers: {}}
+    '''
+    link = BASE_URL + '/details/' + identifier
+    res = requests.get(link)
+    assert(res.status_code == 200)
+    page = soup(res.text)
+    segment = {'snippets': [], 'metadata': {}}
+    # Fetch snippets
+    for column in page.find_all('div', {'class': 'tvcol'}):
+        snippet = {
+            'minute': column.find('div', {'class': 'sniptitle'}).text.strip(),
```

Comment:
I like this new snippet format. It is more clear and allows to add extra data when we want to correlate stocks